user-contrib/Ltac2/tac2env.mli

```diff
@@ -122,6 +122,7 @@ type ('a, 'b) ml_object = {
   ml_subst : Mod_subst.substitution -> 'b -> 'b;
   ml_interp : environment -> 'b -> valexpr Proofview.tactic;
   ml_print : Environ.env -> 'b -> Pp.t;
+  ml_ntn_subst: (Name.t * Glob_term.glob_constr) list -> 'b -> 'b;
```

Comment:
It doesn't make sense to extend the object type declaration when the only object that can appear in a term is the Ltac2 one. I'd simply register by hand the notation substitution function for this entry, and remove all the assertion failures.

Reply:
The thing is I am not confident with these dynamic objects. I’ll give a try to what you’ve suggested.

Reply:
In fact I think I don’t understand what you wrote. Nonetheless, I’ve just pushed a different implementation that does not extend the object type and has no “assert false” in it, but explicitly dispatches on the tag.